Using "allow" condition with no dependency name causes "wrong argument type nil (expected Regexp)" error

**Describe the bug**
If an "allow" condition has no dependency name, an error is thrown when checking for updates.
e.g.

```yml
 - package-ecosystem: "npm"
    allow:
      # Only update production dependencies, do not update developer dependencies
      - dependency-type: "production"
```

The [official docs](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#allow) show examples of conditions without a dependency name, suggesting that when no dependency name is specified, the rule should matches all dependencies.

**Categorization**
- [x] This is not a permissions issue (We cannot solve permission issues)

**Repository**
N/A

**To Reproduce**
Steps to reproduce the behavior:
1. Run Docker updater with `-e DEPENDABOT_ALLOW_CONDITIONS='[{"dependency-type":"production"}]'`

**Expected behavior**
No errors; the rule should match all dependency names.

**Logs**
```log
Checking if @angular-eslint/eslint-plugin-template 16.0.3 needs updating
🌍 --> GET https://registry.npmjs.org/@angular-eslint%2Feslint-plugin-template
🌍 <-- 200 https://registry.npmjs.org/@angular-eslint%2Feslint-plugin-template
I, [2024-07-12T07:19:29.649501 #17]  INFO -- : Filtered out 15 ignored versions
🌍 --> HEAD https://registry.npmjs.org/@angular-eslint/eslint-plugin-template/-/eslint-plugin-template-16.3.1.tgz
🌍 <-- 200 https://registry.npmjs.org/@angular-eslint/eslint-plugin-template/-/eslint-plugin-template-16.3.1.tgz
Requirements to unlock own
Requirements update strategy #<Dependabot::RequirementsUpdateStrategy::BumpVersions>
bin/update_script.rb:238:in `match?': wrong argument type nil (expected Regexp) (TypeError)

  found = $options[:allow_conditions].find { |al| dep.name.match?(al["dependency-name"]) }
                                                                  ^^^^^^^^^^^^^^^^^^^^^
	from bin/update_script.rb:238:in `block in allow_conditions_for'
	from bin/update_script.rb:238:in `each'
	from bin/update_script.rb:238:in `find'
	from bin/update_script.rb:238:in `allow_conditions_for'
	from bin/update_script.rb:620:in `block in <main>'
	from bin/update_script.rb:552:in `each'
	from bin/update_script.rb:552:in `<main>'
##[error]The process '/usr/bin/docker' failed with exit code 1
```

**Extension (please complete the following information):**
 - Host: Azure DevOps
 - Version 1.29.6

**Server (please complete the following information):**
 - Region auseast
 - Version 1.29.6

**Additional context**
N/A

--- a/updater/bin/update_script.rb
+++ b/updater/bin/update_script.rb
@@ -235,7 +235,9 @@ TYPE_HANDLERS = {
 
 def allow_conditions_for(dep)
   # Find where the name matches then get the type e.g. production, direct, etc
-  found = $options[:allow_conditions].find { |al| dep.name.match?(al["dependency-name"]) }
+  found = $options[:allow_conditions].find do |al|
+    Dependabot::Config::UpdateConfig.wildcard_match?(al["dependency-name"] || "*", dep.name)
+  end
   found ? found["dependency-type"] : "all" # when not specified, allow all types
 end
 

